### DISCUSSION TITLE
Pystack_client. perceptions. mirror: The request has exceeded the maximum allowed time

### DISCUSSION THREAD

**Discussion by Ethan-Chen421** [microsoft/PlanetaryComputer#383] [Data request]
`import json
import pystac_client
import planetary_computer
from django.views.decorators.csrf import csrf_exempt
from django.views.decorators.http import require_POST
from django.http import JsonResponse
from urllib.parse import urlparse
@csrf_exempt  
@require_POST
def search_NDVI(request):
    if request.method == 'POST':
        start_date = request.POST.get('start_date')
        end_date = request.POST.get('end_date')
        cloud_cover = request.POST.get('cloud_cover')
        time_of_interest = f'{start_date}/{end_date}'
        geojson_str = request.POST.get('geojson')

        if not geojson_str or not start_date or not end_date:
            return JsonResponse(status=400)

        try:
            geojson = json.loads(geojson_str)
        except json.JSONDecodeError:
            return JsonResponse(status=400)

        if 'features' not in geojson or not geojson['features']:
            return JsonResponse(status=400)

        area_of_interest = geojson['features'][0]['geometry']

        catalog = pystac_client.Client.open(
            "https://planetarycomputer.microsoft.com/api/stac/v1",
            modifier=planetary_computer.sign_inplace,
        )  # pystac_client.

        search = catalog.search(
            collections=["sentinel-2-l2a"],
            intersects=area_of_interest,
            datetime=time_of_interest,
            query={"eo:cloud_cover": {"lt": cloud_cover}},
        )

        items = list(search.get_items())
        if not items:
            return JsonResponse(status=200)
        item_details = []
        for item in items:
           
            item_link = item.assets["visual"].href

            parsed_url = urlparse(item_link)
            filename = parsed_url.path.split('/')[-1]
            filename_without_ext = filename.split('.')[0]

           
            cloud_cover = item.properties['eo:cloud_cover']

            
            item_details.append({
                'Image ID':item.id, 
                'Product parameters': filename_without_ext, 
                'Shooting date': str(item.datetime.date()),
                'Cloud cover': cloud_cover, 
            })

        return JsonResponse({'message': 'successful', 'Data information': item_details})

    return JsonResponse({'error': '只接受POST请求'}, status=405)`
When I talk about the integration of search functionality in Django, the error 'pystack_client. perceptions. error' appears in Django: 'the request exceeded the maximum allowed time', please try again. if the issue persists, please contact  planetarycomputer@microsoft.com.

**Comment by 777arc** [comment]
@Mrjoecen is this still an issue?  there was a brief outage a few weeks ago that may have caused this problem
